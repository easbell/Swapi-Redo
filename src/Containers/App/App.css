.App {
  text-align: center;
  
}

.App-header {
  justify-content: center;
  font-size: 2rem;
  color: #feda4a;
}

.category-header {
  color: #feda4a;
  font-size: 1.7rem;
  text-transform: capitalize;
}