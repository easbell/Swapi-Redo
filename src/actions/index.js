export const setVehicles = (vehicles) => ({
  type: 'ADD_VEHICLES',
  vehicles
})

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasError = (message) => ({
  type: 'HAS_ERROR',
  message
})