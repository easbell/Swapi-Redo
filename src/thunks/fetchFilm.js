import { isLoading, hasError, setFilm } from '../actions'

export const fetchFilm = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(url);
      if(!response.ok) {
        throw Error(response.statusText);
      }
      const data = await response.json();
      const cleanedData = ({openingCrawl: data.opening_crawl, title: data.title, releaseDate: data.release_date})
      dispatch(isLoading(false))
      dispatch(setFilm(cleanedData))
    } catch(error) {
      dispatch(hasError(error.message))
    }
  }
}