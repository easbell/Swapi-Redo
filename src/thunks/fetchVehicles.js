import { isLoading, hasError, setVehicles } from '../actions'
import { cleanVehicles } from '../utils/helpers';

export const fetchVehicles = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(url);
      if(!response.ok) {
        throw Error(response.statusText);
      }
      const data = await response.json();
      const cleanedData = cleanVehicles(data.results);
      dispatch(isLoading(false))
      dispatch(setVehicles(cleanedData))
    } catch(error) {
      dispatch(hasError(error.message))
    }
  }
}