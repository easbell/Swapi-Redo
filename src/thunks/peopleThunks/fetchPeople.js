import { isLoading, hasError, setPeople } from '../../actions';
import { fetchHomeWorld } from './fetchHomeWorld';
import { fetchSpecies } from './fetchSpecies';
import { combineInfo } from '../../utils/helpers';

export const fetchPeople = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(url);
      if(!response.ok) {
        throw Error(response.statusText);
      }
      const data = await response.json();
      const homeworlds = await dispatch(fetchHomeWorld(data.results))
      const species = await dispatch(fetchSpecies(data.results))
      const combined = combineInfo(homeworlds, species)
      dispatch(isLoading(false))
      dispatch(setPeople(combined))
    } catch(error) {
      dispatch(hasError(error.message))
    }
  }
}