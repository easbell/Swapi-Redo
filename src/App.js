import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchFilm } from './thunks/fetchFilm';
import { fetchVehicles } from './thunks/fetchVehicles';
import { fetchPlanets } from './thunks/planetThunks/fetchPlanets';
import { fetchPeople } from './thunks/peopleThunks/fetchPeople';
import FilterControls from './FilterControls';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      randomFilm: {},
      cardsSelected: [],
      errorStatus: ''
    }
  }
  
  async componentDidMount() {
    const random = this.getRandomNum()
    this.props.fetchFilm(`https://swapi.co/api/films/${random}`)
    this.props.fetchVehicles(`https://swapi.co/api/vehicles`)
    this.props.fetchPlanets(`https://swapi.co/api/planets`)
    this.props.fetchPeople(`https://swapi.co/api/people`)
  }

  getRandomNum = () => {
    return Math.floor(Math.random() * (7 - 1) + 1); 
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>SWApi Box</h1>
        </header>
        <div>
          <FilterControls />
          {/* <Splash 
            randomFilm={randomFilm}
          />
          <CardContainer /> */}
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  fetchVehicles: (url) => dispatch(fetchVehicles(url)),
  fetchPlanets: (url) => dispatch(fetchPlanets(url)),
  fetchPeople: (url) => dispatch(fetchPeople(url)),
  fetchFilm: (url) => dispatch(fetchFilm(url))
})

export default connect(null, mapDispatchToProps)(App);
