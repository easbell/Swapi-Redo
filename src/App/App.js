import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchFilm } from '../thunks/fetchFilm';
import { fetchVehicles } from '../thunks/fetchVehicles';
import { fetchPlanets } from '../thunks/fetchPlanets';
import { fetchPeople } from '../thunks/peopleThunks/fetchPeople';
import FilterControls from '../FilterControls/FilterControls';
import CardContainer from '../CardContainer/CardContainer';
import Scroll from '../Scroll/Scroll';
import './App.css';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      pageStatus: 'scroll'
    }
  }

  async componentDidMount() {
    const random = this.getRandomNum()
    this.props.fetchFilm(`https://swapi.co/api/films/${random}`)
    this.props.fetchVehicles(`https://swapi.co/api/vehicles`)
    this.props.fetchPlanets(`https://swapi.co/api/planets`)
    this.props.fetchPeople(`https://swapi.co/api/people`)
  }

  getRandomNum = () => {
    return Math.floor(Math.random() * (7 - 1) + 1); 
  }

  enterApp = () => {
    this.setState({pageStatus: 'home'})
  }

  render() {
    switch(this.state.pageStatus) {
      case('home'):
      return (
        <div className="App">
          <header className="App-header">
            <h1>SWApi Box</h1>
          </header>
          <div>
            <FilterControls />
            <h3 className="category-header">{this.props.category}</h3>
            <CardContainer /> 
          </div>
        </div>
      )
    default: 
      return (
        <div>
          <Scroll enterApp={this.enterApp}/>
        </div>
      )
      }
  }
}

export const mapDispatchToProps = (dispatch) => ({
  fetchVehicles: (url) => dispatch(fetchVehicles(url)),
  fetchPlanets: (url) => dispatch(fetchPlanets(url)),
  fetchPeople: (url) => dispatch(fetchPeople(url)),
  fetchFilm: (url) => dispatch(fetchFilm(url))
})

export const mapStateToProps = (state) => ({
  category: state.category
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
