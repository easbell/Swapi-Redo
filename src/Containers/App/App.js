import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchFilm } from '../../thunks/fetchFilm';
import { fetchVehicles } from '../../thunks/fetchVehicles';
import { fetchPlanets } from '../../thunks/fetchPlanets';
import { fetchPeople } from '../../thunks/peopleThunks/fetchPeople';
import FilterControls from '../FilterControls/FilterControls';
import CardContainer from '../CardContainer/CardContainer';
import Scroll from '../Scroll/Scroll';
import { Route } from 'react-router-dom';
import './App.css';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      pageStatus: 'scroll'
    }
  }

  async componentDidMount() {
    const { fetchFilm, fetchVehicles, fetchPlanets, fetchPeople } = this.props;
    const random = this.getRandomNum()
    fetchFilm(`https://swapi.co/api/films/${random}`)
    fetchVehicles(`https://swapi.co/api/vehicles`)
    fetchPlanets(`https://swapi.co/api/planets`)
    fetchPeople(`https://swapi.co/api/people`)
  }

  getRandomNum = () => {
    return Math.floor(Math.random() * (7 - 1) + 1); 
  }

  enterApp = () => {
    this.setState({pageStatus: 'home'})
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>SWApi Box</h1>
        </header>
        <Route exact path="/" component={Scroll}/>
        <Route exact path='/main' render={() => (
          <div>
            <FilterControls />
            <h3 className="category-header">{this.props.category}</h3>
            <CardContainer /> 
          </div>
        )} />
      </div>
    )
  }
} 

export const mapDispatchToProps = (dispatch) => ({
  fetchVehicles: (url) => dispatch(fetchVehicles(url)),
  fetchPlanets: (url) => dispatch(fetchPlanets(url)),
  fetchPeople: (url) => dispatch(fetchPeople(url)),
  fetchFilm: (url) => dispatch(fetchFilm(url))
})

export const mapStateToProps = (state) => ({
  category: state.category
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
