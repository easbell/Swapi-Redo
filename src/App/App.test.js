import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapDispatchToProps, mapStateToProps } from './App';
import { setVehicles, setFilm, setPlanets, setPeople } from '../actions';
import { fetchFilm } from '../thunks/fetchFilm';
import { fetchVehicles } from '../thunks/fetchVehicles';
import { fetchPlanets } from '../thunks/fetchPlanets';
import { fetchPeople } from '../thunks/peopleThunks/fetchPeople';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with state', () => {
    wrapper.setState({pageStatus: 'home'});
    expect(wrapper).toMatchSnapshot();
  })

  it('should have default state', () => {
    expect(wrapper.state('pageStatus')).toEqual('scroll');
  });

  describe('componentDidMount', () => {
    it('should call fetch Thunks when mounted', () => {
      const mockFn = jest.fn();
      const wrapper = shallow(
        <App 
          fetchFilm={mockFn} 
          fetchVehicles={mockFn}
          fetchPlanets={mockFn}
          fetchPeople={mockFn}
        />);
      const instance = wrapper.instance();
      instance.componentDidMount();

      expect(mockFn).toHaveBeenCalled();
    });
  });

  describe('enterApp', () => {
    it('should set state when enterApp is invoked', () => {
      wrapper.setState({pageStatus: 'scroll'});
      wrapper.instance().enterApp();

      expect(wrapper.state('pageStatus')).toEqual('home');
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with category', () => {
      const mockState = {
        vehicles: ['vehicle', 'car'],
        category: 'planets'
      };

      const expectedState = {
        category: 'planets'
      };

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedState);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a fetchFilm action', async () => {
      const dispatch = jest.fn();
      const mockUrl = 'www.website.com';
      const mockThunk = await fetchFilm(mockUrl);
      const actionToDispatch = setFilm(mockThunk)
      
      mapDispatchToProps(dispatch(setFilm(mockThunk)))
      expect(dispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

  it('calls dispatch with a fetchVehicles action', async () => {
    const dispatch = jest.fn();
    const mockUrl = 'www.website.com';
    const mockThunk = await fetchVehicles(mockUrl);
    const actionToDispatch = setVehicles(mockThunk)
    
    mapDispatchToProps(dispatch(setVehicles(mockThunk)))
    expect(dispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('calls dispatch with a fetchPlanets action', async () => {
    const dispatch = jest.fn();
    const mockUrl = 'www.website.com';
    const mockThunk = await fetchPlanets(mockUrl);
    const actionToDispatch = setPlanets(mockThunk)
    
    mapDispatchToProps(dispatch(setPlanets(mockThunk)))
    expect(dispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('calls dispatch with a fetchPeople action', async () => {
    const dispatch = jest.fn();
    const mockUrl = 'www.website.com';
    const mockThunk = await fetchPeople(mockUrl);
    const actionToDispatch = setPeople(mockThunk)
    
    mapDispatchToProps(dispatch(setPeople(mockThunk)))
    expect(dispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});