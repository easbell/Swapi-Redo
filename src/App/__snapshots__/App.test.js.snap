// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(Scroll)
        enterApp={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "enterApp": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(Scroll)
          enterApp={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "enterApp": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`App matches snapshot with state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          <h1>
            SWApi Box
          </h1>
        </header>,
        <div>
          <Connect(FilterControls) />
          <h3
            className="category-header"
          />
          <Connect(CardContainer) />
        </div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            SWApi Box
          </h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWApi Box",
          },
          "ref": null,
          "rendered": "SWApi Box",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(FilterControls) />,
            <h3
              className="category-header"
            />,
            <Connect(CardContainer) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "category-header",
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            <h1>
              SWApi Box
            </h1>
          </header>,
          <div>
            <Connect(FilterControls) />
            <h3
              className="category-header"
            />
            <Connect(CardContainer) />
          </div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              SWApi Box
            </h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SWApi Box",
            },
            "ref": null,
            "rendered": "SWApi Box",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(FilterControls) />,
              <h3
                className="category-header"
              />,
              <Connect(CardContainer) />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "category-header",
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
